### PR TITLE
Updated the register alert source feature to support API changes. Added Delete Alert source.

### LINKED ISSUE #384: [Delfin UI] - Register Alert source and Clear Alert Source
**Issue/Feature Description:**

- [x] Register an alert source on a storage array
- [x] Update alert source on storage array.
- [x] Clear the alert source on storage array

**Why this issue to fixed / feature is needed(give scenarios or use cases):**
The user should be able to register the alert source for the storage in order to generate the alerts. In case the alert source needs to be changed the user should be able to remove the old one and register a new one.

**How to reproduce, in case of a bug:**

**Other Notes /  Environment Information: (Please give the env information, log link or any useful information for this issue)**

### PR DESCRIPTION
**Pull request by anvithks**
**What type of PR is this?**
/kind new feature
/kind enhancement

**What this PR does / why we need it**:
This PR updates the Register Alert source feature.
The API spec is changed and the same is now reflected in the form.
The user can now modify / register an alert source using the same form. 
This form is now called `Configure Alert Source` instead of register.
User can choose between different versions and each version has specific parameters.

This PR also contains some enhancements to the context menu and the overall storage details code.
Removed some unused methods and modified some existing ones.
**Which issue(s) this PR fixes**:
Fixes #384 
- [x] Register an alert source on a storage array
- [x] Update alert source on storage array.
- [x] Clear the alert source on storage array

**Test Report Added?**:
/kind TESTED

**Test Report**:

**Configure Alert Source Form**
configure-form

**Configure Alert Source SNMPV2 Fields**
configure-form-snmpv2c

**Configure Alert Source SNMPV2 Fields**
configure-form-snmpv2c-fields

**Configure Alert Source SNMPV2 Success**
configure-form-snmpv2c-success

**Configure Alert Source SNMPV3 Fields**
configure-form-snmpv3-all-fields

**Configure Alert Source SNMPV2 Auth Protocols**
configure-form-snmpv3-fields

**Configure Alert Source SNMPV2 Privacy Protocols**
configure-form-snmpv3-privay-protocol-options

**Delete Alert source**
delete-alert-source

**Delete Alert source Not exist**
delete-alert-source-not-exist

**Special notes for your reviewer**:

### CODE REVIEW

File: src/app/business/delfin/storages/storages.component.ts [hunk 1]
```diff
@@ -592,16 +656,29 @@ export class StoragesComponent implements OnInit {
...
     }
-    
-
-    
-
-    
-
-    lazyLoadTreeChildren(event){
 
+   getVolumesByStorage(storageId){
```
Comment (kumarashit) [inline src/app/business/delfin/storages/storages.component.ts:660]: These changes are not for this PR, right?

File: src/app/business/delfin/storages/storages.component.ts [hunk 2]
```diff
@@ -592,16 +656,29 @@ export class StoragesComponent implements OnInit {
...
+        this.ds.getAllVolumes(storageId).subscribe((res)=>{
+            let vols = res.json().volumes;
+           return vols;
+        }, (error)=>{
+            console.log("Something went wrong. Could not fetch Volumes.", error)
+        });
+       
     }
 
+    getPoolsByStorage(storageId){
```
Comment (kumarashit) [inline src/app/business/delfin/storages/storages.component.ts:671]: These changes are not for this PR, right?

File: src/app/business/delfin/storages/storages.component.ts (hunk 1 above)
Comment (anvithks) [reply src/app/business/delfin/storages/storages.component.ts:660]: These are helper functions that I have created .
Not for this feature but part of the enhancement.

File: src/app/business/delfin/storages/storages.component.ts (hunk 2 above)
Comment (anvithks) [reply src/app/business/delfin/storages/storages.component.ts:671]: These are helper functions that I have created .
Not for this feature but part of the enhancement.

**Review by kumarashit — APPROVED** [review]
LGTM

**Review by skdwriting — APPROVED** [review]
LGTM

**Pull request merged**